setup.py

```diff
@@ -10,7 +10,7 @@
 
 # at least 1.3 to be able to use XMLDataSet and pandas integration with fsspec
 PANDAS = "pandas~=1.3"
-SPARK = "pyspark>=2.2, <4.0"
+SPARK = "pyspark>=2.2, <3.4"
```

Comment:
Set a strict lower cap because 3.4 breaks a lot of things, it's very hard to fix it without changing implementation. We shouldn't do it in `kedro`, will leave this to `kedro-datasets` instead.